graphium/utils/tensor.py

```diff
@@ -179,7 +179,7 @@ def nan_median(input: Tensor, **kwargs) -> Tensor:
                 median = median.squeeze(d)
     else:
         if dim is None:
-            median = torch.nanmedian(input.flatten())
+            median = torch.nanmedian(input.flatten().float())
```

Comment:
`torch.nanmedian` not implemented for `float16`

Reply:
This will change the dtype of the output. I would suggest casting back the `median` to its original dtype after the call to `.float()`